src/ts/services/transactions/index.ts

```diff
@@ -0,0 +1,29 @@
+import { Network } from '../../constants/networks'
+import { ApiPromise, WsProvider } from '@polkadot/api'
+import { KeyringPair$Json } from '@polkadot/keyring/types'
+import Keyring from '@polkadot/keyring'
+
+export const transfer = async (
+  amount: number,
+  keyringPair$Json: KeyringPair$Json,
+  toAddress: string,
+  network: Network): Promise<string> => {
+
+  const provider = new WsProvider(network.rpcServer)
+
+  // Instantiate the API
+  const api = await ApiPromise.create(provider)
+
+  const keyring = new Keyring({ type: 'sr25519' })
+
+  const from = keyring.addFromJson(keyringPair$Json)
```

Comment:
if we are going to use `addFromJson`, I think we would need to check if the key is locked and ask for password to unlock it.

Reply:
yeah. good point

Reply:
you can add logic:
```node
if(keyring.islocked()) {
// ask for pw
}

Reply:
I will probably move the transaction signing logic in keyring-vault. I will push a change later to demonstrate it.